From 8391dc3891337ecff8becefe1927a30d3922b848 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 1 May 2024 16:50:50 +0200
Subject: [PATCH] res/stasis/control.c: include signal.h

Include signal.h to avoid the following build failure with uclibc-ng
raised since
https://github.com/asterisk/asterisk/commit/2694792e13c7f3ab1911c4a69fba0df32c544177:

stasis/control.c: In function 'exec_command_on_condition':
stasis/control.c:313:3: warning: implicit declaration of function 'pthread_kill'; did you mean 'pthread_yield'? [-Wimplicit-function-declaration]
  313 |   pthread_kill(control->control_thread, SIGURG);
      |   ^~~~~~~~~~~~
      |   pthread_yield
stasis/control.c:313:41: error: 'SIGURG' undeclared (first use in this function)
  313 |   pthread_kill(control->control_thread, SIGURG);
      |                                         ^~~~~~

Fixes:
 - http://autobuild.buildroot.org/results/d16e4ca4bd26234f84d17da24c04a8c19faba6c5

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: https://github.com/asterisk/asterisk/pull/728
---
 res/stasis/control.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/res/stasis/control.c b/res/stasis/control.c
index 360f20790..4ed0e2af1 100644
--- a/res/stasis/control.c
+++ b/res/stasis/control.c
@@ -41,6 +41,8 @@
 #include "asterisk/musiconhold.h"
 #include "asterisk/app.h"
 
+#include <signal.h>
+
 AST_LIST_HEAD(app_control_rules, stasis_app_control_rule);
 
 /*!
-- 
2.43.0

